spec/helpers.lua

```diff
@@ -1554,6 +1561,129 @@ local function wait_for_invalidation(key, timeout)
 end
 
 
+--- Wait for all targets, upstreams, services, and routes update
+-- 
+-- NOTE: this function is not available for DBless-mode
+-- @function wait_for_all_config_update
+-- @tparam[opt=30] number timeout maximum time to wait
+-- @tparam[opt] number admin_client_timeout, to override the default timeout setting
+-- @tparam[opt] number forced_admin_port to override the default port of admin API
+-- @usage helpers.wait_timer()
+local function wait_for_all_config_update(timeout, admin_client_timeout, forced_admin_port)
+  timeout = timeout or 30
+
+  local client = admin_client(admin_client_timeout, forced_admin_port)
+
+  local upstream_id, target_id, service_id, route_id
+  local upstream_name = "really.really.really.really.really.really.really.mocking.upstream.com"
+  local service_name = "really-really-really-really-really-really-really-mocking-service"
+  local route_path = "/really-really-really-really-really-really-really-mocking-route"
+  local host = MOCK_UPSTREAM_HOST
+  local port = MOCK_UPSTREAM_PORT
+
+  local res = client:post("/upstreams", {
+    headers = {["Content-Type"] = "application/json"},
+    body = { name = upstream_name },
+  })
+
+  local body = luassert.res_status(201, res)
+  local json = cjson.decode(body)
+
+  if not json.id then
+    error("failed to add dummy upstream")
+  end
+
+  upstream_id = json.id
+
+  -----------------------------
```

Comment:
This line could be more meaningful, such as `-- create a new target`.

Reply:
done